multi_partner_learning.py

```diff
@@ -67,9 +61,11 @@ def __init__(self,
         self.aggregation_weights = []
         self.models_weights_list = [None] * self.partners_count
         self.local_score_list = [None] * self.partners_count
-        self.score_matrix_minibatch_end = np.zeros(shape=(self.epoch_count, self.minibatch_count, self.partners_count))
-        self.score_matrix_minibatch_start = np.zeros(shape=(self.epoch_count, self.minibatch_count))
-        self.global_val_acc, self.global_val_loss = [], []
+        self.score_matrix_per_partner = np.zeros(shape=(self.epoch_count, self.minibatch_count, self.partners_count))
+        self.score_matrix_collective_models = np.zeros(shape=(self.epoch_count, self.minibatch_count + 1))
```

Comment:
Why `+ 1` ?

Reply:
We save it at beginning of each mini-batch, plus at the end of the last of each epoch.